platform/src/plot.rs

```diff
@@ -0,0 +1,81 @@
+use crate::config::Config;
+use plotters::coord::{Shift};
+use plotters::prelude::*;
+
+pub fn plot(config: &Config) -> Result<(), Box<dyn std::error::Error>> {
+    let isSvg = config.outputFilePath.as_str().ends_with(".svg");
+    if isSvg {
+        let backend = construct_svg_backend(config);
+        plot_with(backend, config);
+    } else {
+        let backend = construct_bitmap_backend(config);
+        plot_with(backend, config);
+    }
+    Ok(())
+}
+
+fn construct_area<Backend: DrawingBackend>(
+    backend: Backend,
+    config: &Config,
+) -> Result<
+    DrawingArea<Backend, Shift>,
+    DrawingAreaErrorKind<Backend::ErrorType>,
+> {
+    let area = backend.into_drawing_area();
+    area.fill(&WHITE)?;
+    let area = area.titled(config.title.as_str(), ("sans-serif", 60))?;
+    Ok(area)
+}
+
+fn construct_bitmap_backend(config: &Config) -> BitMapBackend {
+    BitMapBackend::new(
+        config.outputFilePath.as_str(),
+        (config.width, config.height),
+    )
+}
+
+fn construct_svg_backend(config: &Config) -> SVGBackend {
+    SVGBackend::new(
+        config.outputFilePath.as_str(),
+        (config.width, config.height),
+    )
+}
+
+fn plot_with<Backend: DrawingBackend>(
+    backend: Backend,
+    config: &Config,
+) -> Result<(), DrawingAreaErrorKind<Backend::ErrorType>> {
+    let area = construct_area(backend, config)?;
+    println!("x: {:?}", config.x);
```

Comment:
nit: switch from `println!` to `dbg!` Then you don't need to specify the variable name, get a line number, and can even do it inline where it makes sense. ex:
```rust
let xy1: Vec<(i32, i32)> = dbg!(config.x.into_iter().zip(config.y1.into_iter()).collect());
```

Reply:
That's awesome!

Reply:
Added in a new commit, thanks